cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -405,155 +406,149 @@ runQueryKesPeriodInfo (AnyConsensusModeParams cModeParams) network nodeOpCertFil
-     -- our KES starting period is within that interval.
-
-     let slotsPerKesPeriod = fromIntegral protocolParamSlotsPerKESPeriod
-         maxKesEvolutions  = fromIntegral protocolParamMaxKESEvolutions
-         currentKesPeriod = unSlotNo currSlot `div` slotsPerKesPeriod
-
-         kesPeriodIntervalStart = (currentKesPeriod `div` maxKesEvolutions) * maxKesEvolutions
-         kesPeriodIntervalEnd = kesPeriodIntervalStart + maxKesEvolutions
-
-         opCertKesPeriod = fromIntegral $ getKesPeriod opCert
-
-         kesPeriodIntervalEndInSlots = kesPeriodIntervalEnd * slotsPerKesPeriod
-
-         slotsTillNewKesKey = kesPeriodIntervalEndInSlots - unSlotNo currSlot
-
-        -- See OCERT rule in ledger specs
-     if kesPeriodIntervalStart <= opCertKesPeriod &&
-        -- Checks if op cert KES period is less than the interval start
-        opCertKesPeriod < kesPeriodIntervalEnd
-        -- Checks if op cert KES period is less than the interval end
-     then
-       return ( slotsTillNewKesKey
-              , currentKesPeriod
-              , kesPeriodIntervalStart
-              , kesPeriodIntervalEnd
-              , SuccessDiagnostic OpCertCurrentKesPeriodWithinInterval
-              )
-     else do
-       let fd = FailureDiagnostic $ OpCertKesPeriodOutsideOfCurrentInterval
-                                      (nodeOpCertFile, opCertKesPeriod)
-                                      kesPeriodIntervalStart
-                                      kesPeriodIntervalEnd
-       return ( 0
-              , currentKesPeriod
-              , kesPeriodIntervalStart
-              , kesPeriodIntervalEnd
-              , fd
-              )
+     -> OpCertEndingKesPeriod
+     -> Maybe UTCTime
+   opCertExpiryUtcTime eInfo gParams (OpCertEndingKesPeriod oCertExpiryKesPeriod) =
+     let time = epochInfoSlotToUTCTime
+                  eInfo
+                  (SystemStart $ protocolParamSystemStart gParams)
+                  (fromIntegral $ oCertExpiryKesPeriod * fromIntegral (protocolParamSlotsPerKESPeriod gParams))
+     in case time of
+          Left _ -> Nothing
+          Right t -> Just t
+
+   renderOpCertNodeAndOnDiskCounterInformation :: FilePath -> OpCertNodeAndOnDiskCounterInformation -> IO ()
+   renderOpCertNodeAndOnDiskCounterInformation opCertFile opCertCounterInfo =
+     case opCertCounterInfo of
+      OpCertOnDiskCounterMoreThanOrEqualToNodeState _ _ ->
+        putStrLn @String "✓ The operational certificate counter agrees with the node protocol state counter"
+      OpCertOnDiskCounterBehindNodeState onDiskC nodeStateC ->
+        putStrLn $ "✗ The protocol state counter is greater than the counter in the operational certificate at: " <> opCertFile <> "\n" <>
+                   "On disk operational certificate counter: " <> show (unOpCertOnDiskCounter onDiskC) <> "\n" <>
+                   "Protocol state counter: " <> show (unOpCertNodeStateCounter nodeStateC)
```

Comment:
Perhaps we should indent this lines so the output looks like this to aid readability?

```
✗ The protocol state counter is greater than the counter in the operational certificate at: ...
    On disk operational certificate counter: ...
    Protocol state counter: ...
```

Similarly below.